src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationService.java

```diff
@@ -346,6 +328,19 @@ private String getSubscriptionDetails(String newSubscription, String oldSubscrip
         return subscription;
     }
 
+    boolean isNotificationStillValidToSendSetAsideRequest(SscsCaseData caseData, NotificationEventType eventType) {
+        List<String> originalSenders = Arrays.asList("dwp", "hmcts");
+        switch (eventType) {
+            case ACTION_FURTHER_EVIDENCE:
+            case POST_HEARING_REQUEST:
+                return !(isNull(caseData.getOriginalSender())
+                    || originalSenders.contains(caseData.getOriginalSender().getValue().getCode()));
```

Comment:
caseData.getOriginalSender().getValue().getCode() thats NPE potential
if caseData.getOriginalSender()  is null

and because in the statement you use OR (||) the right part will always be checked

Reply:
Checked it in debugger and its working as intended